fairchive-webapp/src/test/java/edu/harvard/iq/dataverse/datafile/FileIntegrityCheckerTest.java

```diff
@@ -167,6 +167,25 @@ public void checkFilesIntegrity__with_storage_md5_different_than_datafile_md5()
         assertEquals(datafile1, integrityReport.getSuspicious().get(0).getIntegrityFailFile());
     }
 
+    @Test
+    public void checkFilesIntegrity__with_no_storage_md5() throws IOException {
+        // given
+        DataFile datafile1 = makeDataFile(102L, null, null);
+
+        StorageIO<DataFile> datafileStorage1 = mockDataFileStorageIO(true, 102L, true, "md5hash_other");
```

Comment:
From the test name I would assume that it should be the other way around. In `makeDataFile` we setup a md5, and in `mockDataFileStorageIO` we don't.